mlserver/parallel/pool.py

```diff
@@ -66,19 +93,80 @@ def env_hash(self) -> Optional[str]:
 
         return self._env.env_hash
 
+    @property
+    def name(self) -> str:
+        if self.env_hash:
+            return f"inference pool with hash '{self.env_hash}'"
+
+        return "default inference pool"
+
+    async def on_worker_stop(self, pid: int, exit_code: int):
+        if pid not in self._workers:
+            # If this worker didn't belong to this pool, ignore
+            return
+
+        worker = self._workers[pid]
+        logger.warning(
+            f"Worker with PID {worker.pid} on {self.name} stopped "
+            f"unexpectedly with exit code {exit_code}. "
+            "Triggering worker restart..."
+        )
+        self._dispatcher.on_worker_stop(worker, exit_code)
+        if pid in self._workers:
+            # NOTE: worker may be removed by dispatcher
+            del self._workers[pid]
+
+        # Call attached on_worker_stop hooks
+        await asyncio.gather(*[callback(worker) for callback in self._on_worker_stop])
+
+        # Start a new worker
+        await self._start_worker()
```

Comment:
is there any race condition when we start a new worker and any new model load calls?